### PR TITLE
[16.0][ADD] account_invoice_refund_code

### PR DESCRIPTION
**Pull request by OriolMForgeFlow**
This module allows to have specific refund sequences.

### CODE REVIEW

File: account_invoice_refund_code/README.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_refund_code
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runboat-Try%20me-875A7B.png
+    :target: https://runboat.odoo-community.org/builds?repo=OCA/account-invoicing&target_branch=16.0
+    :alt: Try me on Runboat
+
+|badge1| |badge2| |badge3| |badge4| |badge5|
+
+This module allows to have specific refund sequences.
```
Comment (JordiMForgeFlow) [inline account_invoice_refund_code/README.rst]:
```suggestion
This module allows to have specific refund codes.
```

File: account_invoice_refund_code/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+class AccountMove(models.Model):
+    _name = "account.move"
+    _inherit = ["account.move"]
+
+    def _get_starting_sequence(self):
+        starting_sequence = super()._get_starting_sequence()
+
+        if self.journal_id.refund_sequence and self.journal_id.refund_code:
+            if starting_sequence:
```
Comment (JordiMForgeFlow) [inline account_invoice_refund_code/models/account_move.py]:
```suggestion
        if self.journal_id.refund_sequence and self.journal_id.refund_code and starting_sequence:
```

File: account_invoice_refund_code/views/account_journal_views.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" ?>
+<odoo>
+    <record id="view_account_journal_form" model="ir.ui.view">
+        <field name="name">account.journal.form - account_invoice_refund_code</field>
+        <field name="model">account.journal</field>
+        <field name="inherit_id" ref="account.view_account_journal_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='code']" position="after">
+                <field name="refund_code" />
```
Comment (JordiMForgeFlow) [inline account_invoice_refund_code/views/account_journal_views.xml]: we could make the field invisible if you do not choose to have a different sequence for refunds. Also we should hide the field in the same way it is done for the refund sequence (`attrs="{'invisible': [('type', 'not in', ['sale', 'purchase'])]}"`)

**Comment by OriolMForgeFlow** [comment]
@JordiMForgeFlow  All comments attended

**Review by JordiMForgeFlow — APPROVED** [review]
Code review LGTM 👍🏼

**Review by AaronHForgeFlow — COMMENTED** [review]
LGTM, Functional review

**Review by SergiCForgeFlow — APPROVED** [review]
LGTM :+1:

**Review by JoanSForgeFlow — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]

**Pull request merged**